### DISCUSSION TITLE
[Discussion] AdBlock over VPN

### DISCUSSION THREAD

**Discussion by M86xKC** [Mygod/VPNHotspot#216] [Q&A] [answered]
What are the 'dummy' vpns (adblock over vpn) that you tested with the app and worked?

I have tested many and none of them work!

Can you please give some hints?

Thanks

**Comment by Mygod** [comment, marked as answer]
Here is a list of "dummy VPN" apps (i.e. apps that do not forward most of your traffic elsewhere) that have been to reported to work. Feel free to add more in this thread. Note that these VPNs may not hide your traffic against your carrier (and may still limit your bandwidth for video streams for example) so it is not as perfect as a real VPN (if you are looking for something for free, you can use Warp in [1.1.1.1](https://play.google.com/store/apps/details?id=com.cloudflare.onedotonedotonedotone), which is a free lite VPN instead of dummy but has limited NAT support).

* [AdGuard for Android](https://adguard.com/en/adguard-android/overview.html) (not the version on Google Play)
* [1.1.1.1](https://play.google.com/store/apps/details?id=com.cloudflare.onedotonedotonedotone) without Warp; credits: https://github.com/Mygod/VPNHotspot/discussions/216#discussioncomment-671604
* TrackerControl, credits: https://github.com/Mygod/VPNHotspot/discussions/216#discussioncomment-3612993
* NetGuard, credits: https://github.com/Mygod/VPNHotspot/discussions/216#discussioncomment-3679515
* RethinkDNS, credits: https://github.com/Mygod/VPNHotspot/discussions/216#discussioncomment-3679489 ([might not work perfectly since v055](https://github.com/Mygod/VPNHotspot/discussions/216#discussioncomment-6759738))
* To test if your favorite dummy VPN works: Go into settings and you should see the upstreams listed in "Current upstreams". If the primary/VPN upstream is displayed in bold, it should be good for bypassing tethering limitations. (https://github.com/Mygod/VPNHotspot/issues/139#issuecomment-576441912)

Not working:

* AdAway v6.1.1
* Blokada 4 & 5

tl;dr: Use Warp in [1.1.1.1](https://play.google.com/store/apps/details?id=com.cloudflare.onedotonedotonedotone).

**Reply by ignoramous** [reply]
What's the context here. I'd very much like to see if the app I co-develop [RethinkDNS + Firewall](https://github.com/celzero/rethink-app) works too. Thanks.

Disclosure: I've had some interesting interaction with the Blokada team: https://archive.is/X4nOS | https://archive.is/intdM but that's not why I am here (:

**Reply by Mygod** [reply]
@ignoramous You can take a look at https://github.com/Mygod/VPNHotspot/commit/cce3223311134618446db6aec4cba5548a3d7dd6. Essentially as long as your VPN service intercepts all traffic (and forwards it as normal traffic), it should be good for bypassing tethering restrictions.

**Comment by M86xKC** [comment]
The first one does not use VPN API it only works with Samsung & Yandex browser.

I tried blokada, couldn't get it to work. I tried many others but no chance.

NetGuard works but there are many network drops unfortunately

**Comment by Mygod** [comment]
I remember you can make the first one use VPN by tweaking settings. What do you mean exactly by not working? What are your network provider's restrictions on tethering?

**Comment by M86xKC** [comment]
> What do you mean exactly by not working?

I can't access the internet with hotspot.

But when I use a proper vpn I can using VPNHotspot

**Comment by jmacul2** [comment]
More a DNS VPN, but free:

1.1.1.1 - https://play.google.com/store/apps/details?id=com.cloudflare.onedotonedotonedotone
Make sure you are just using DNS and NOT warp.

**Comment by Mygod** [comment]
Alright I did a little testing. It turns out that [this VPN version of AdGuard](https://adguard.com/en/adguard-android/overview.html) works but not Blokada. This is because Blokada basically works by set up a separate DNS server only so all traffic are still routed directly.

I could set up some diagnosis option to help you find which adblocking VPN would work.

**Comment by Mygod** [comment]
Please try v2.7.0. Go into settings and you should see the upstreams listed in "Current upstreams". If the primary/VPN upstream is displayed in bold, it should be good for bypassing tethering limitations.

**Comment by androidacy-user** [comment]
AdGuard VPN works great with VPN tether, blokada does too but ONLY with it's built in (paid) VPN turned on (which slows the connection whereas AdGuard doesn't).
I'm using DNS over TLS with AdGuard (set up in app) so that may help

**Comment by worstperson** [comment]
AdGuard and AdClear both well as full local VPN implementations. 1.1.1.1 did not work for me unless Warp is enabled, and I found the WireGuard app to be more consistent for that. Host blockers seem to only implement enough to forge DNS responses, so they usually don't work either.

**Comment by puertajulian** [comment]
Versión 6.6 of 1.1.1.1 bring leaks in my case. Just if you have time to check.  Had to revert to 6.5 for it to stop leaking.

**Comment by SuspectTyrannosaurus** [comment]
Thank you @puertajulian for mentioning your trouble with Cloudflare / 1.1.1.1.

My hotspot stopped working (clients could connect but not access the internet), but reverting to version 6.5 of the 1.1.1.1 app (which I got from apkmirror) fixed it.

Neither version 6.6 nor 6.7 of "1.1.1.1" worked for me.

@Mygod :
Since an update to 1.1.1.1 broke things, I'm guessing the trouble isn't with VPNHotspot? I'm happy to help troubleshoot if there are logs I can provide or anything.

**Comment by worstperson** [comment]
Not an issue with VPNHotspot. If you have issues with 1.1.1.1 I suggest generating a profile with wgcf and importing it into the official Wireguard app. 
https://github.com/ViRb3/wgcf
https://play.google.com/store/apps/details?id=com.wireguard.android&hl=en_US&gl=US
Doing this let's me leave it as an always-on VPN without any network interruptions. Paid Warp+ will disconnect every 24hrs due to an unsupported feature(afaict) though.

**Reply by Dervun** [reply]
So what do you suggest? I generated a config with ViRb3/wgcf, imported it to the WireGuard app. Now I have "WARP VPN" on my smartphone, but I already had it with just the 1.1.1.1 official app. How can I do tethering from this device without leaving the VPN? So that after connection to this mobile hotspot, the devices were in the VPN too.
I tried 1.1.1.1 official app + VPNHotspot, WireGuard + VPNHotspot. I disabled Tethering hardware acceleration in VPNHotspot, "allowed remote control apps" in WireGuard. But when I turn on VPN in WireGuard, and then in VPNHotspot I turn on the Wi-Fi hotspot, touch the eye on the top (turned on "swlan0"), nothing happens. VPN still there is on the smartphone and there isn't on connected to the hotspot devices.
Maybe I should set up something in the settings of VPNHotspot (Upstream network interface, Fallback upstream interface). But I don't know what they do, how it works, and what exactly should I set there.
I've already wasted 2 days solving this issue. Maybe there are some articles about setting it up, or how it works.
Help :_(

**Reply by worstperson** [reply]
Sorry, but I am having a bit of trouble parsing the question.

The only issue specific to Wireguard that I know of is when you have the kernel module enabled. You have to set the upstream connection in VPNHotspot to the Wireguard profile name in this case.

If your not able to get VPNHotspot to work with any VPN, post a new issue with information about your device, rom, and logs(logcat/debug info).

**Comment by Mygod** [comment]
I am converting this issue to a discussion given the interest. 😉

**Comment by famewolf** [comment]
Anyone tried trackercontrol from f-droid?

**Reply by Mokura** [reply]
> Anyone tried trackercontrol from f-droid?

This is what I get with TrackerControl with default settings on a Pixel 5a with LineageOS 19.1.

![Screenshot_20220909-235655_VPN Hotspot](https://user-images.githubusercontent.com/5167919/189467800-506a583d-821d-4410-98cb-5320bd2b49c8.png)

**Reply by Mokura** [reply]
After some testing, it appears to work - data use is attributed to TC rather than tethering.

**Reply by Mokura** [reply]
NetGuard also works, as it's what TC is based on.  However, you need to go in to advanced options and turn on "Filter traffic" for `tun0` to appear.

**Reply by MrEngineerMind** [reply]
I installed TC and had tun0 appear in bold in VPN Hotspot, but tether data was still detected on AT&T (Android 10 rooted).

**Comment by Mokura** [comment]
RethinkDNS (https://github.com/celzero/rethink-app) shows `tun1` in bold on the upstream list while in use, regardless of what mode it's in.  (fw+dns, dns only, fw only)

**Reply by ignoramous** [reply]
Just a nit, Rethink is no longer a "dummy VPN" since `v055`. It can forward TCP/UDP to multiple WireGuard upstreams and Tor. At the same time...

**Reply by Mygod** [reply]
Well it still *can* be a dummy VPN? :)

**Comment by yukichigai** [comment]
While current versions of Blokada (5 and 6) do not work, I have had success with Blokada 4. Most up-to-date version is 4.4.15.0. Tested on a rooted Razer Phone 2 running Android 9, so I can't confirm if this works on later versions of Android.

**Reply by Mygod** [reply]
Just tested and Blokada 4 seems identical to 5 on my end.

**Comment by G-cell-coder** [comment]
While I set DNS as 1.1.1.1 - I am still getting ads with youtube and mobile app notifications. Is there any possibility to streamline  the browser to allow the restricted categories of websites through enabling DNS-server detail or VPN details over browser. If any options are available, please update
